ecosystem/web-wallet/src/pages/Wallet.tsx

```diff
@@ -1,20 +1,224 @@
+  const [accountResources, setAccountResources] = useState<AccountResource[] | undefined>(undefined)
+  const [refreshState, setRefreshState] = useState(true)
+  const [isFaucetLoading, setIsFaucetLoading] = useState(false)
+  const [isTransferLoading, setIsTransferLoading] = useState(false)
+
+  const address = aptosAccount?.address().hex()
+  const accountResource = (accountResources) ? accountResources?.find((r) => r.type === '0x1::TestCoin::Balance') : undefined
+  const tokenBalance = (accountResource) ? (accountResource.data as { coin: { value: string } }).coin.value : undefined
+  const tokenBalanceString = numeral(tokenBalance).format('0,0.0000')
+  const toAddress: string | undefined | null = watch('toAddress')
+  const transferAmount: string | undefined | null = watch('transferAmount')
+
+  const onSubmit: SubmitHandler<Inputs> = async (data, event) => {
+    event?.preventDefault()
+    if (toAddress && aptosAccount && transferAmount) {
+      setIsTransferLoading(true)
+      await submitTransaction({
+        toAddress,
+        fromAddress: aptosAccount,
+        amount: transferAmount
+      })
+      setRefreshState(!refreshState)
+      setIsTransferLoading(false)
+      onClose()
+    }
+  }
+
+  const faucetOnClick = async () => {
+    setIsFaucetLoading(true)
+    await fundWithFaucet({ address })
+    setRefreshState(!refreshState)
+    setIsFaucetLoading(false)
+  }
+
+  useEffect(() => {
+    getAccountResources({ address })?.then((data) => {
+      const tempAccountResources = data?.data
+      setAccountResources(tempAccountResources)
+    })
+  }, [refreshState])
 
-export default function Wallet () {
-  const [state] = useGlobalState()
-  const address = state.account?.address().hex()
-  const pubKey = state.account?.pubKey().hex()
   return (
-    <div className="App-header">
-      <h2>Aptos Wallet </h2>
-      <p>Address: {address}</p>
-      <p>Public Key: {pubKey}</p>
-    </div>
+    <VStack
+      height="100%"
+      maxW="100%"
+      width="100%"
+      bgColor={secondaryBgColor[colorMode]}
+    >
+      <WalletHeader />
+      <VStack width="100%" paddingTop={8}>
+        <Text fontSize="sm" color={seconaryAddressFontColor[colorMode]}>Account balance</Text>
+        <Heading>{tokenBalanceString}</Heading>
+        <HStack spacing={4}>
+          <Button
+            isLoading={isFaucetLoading}
+            leftIcon={<FaFaucet />}
+            onClick={faucetOnClick}
+            isDisabled={isFaucetLoading}
+          >
+            Faucet
+          </Button>
+          <Popover
+            isOpen={isOpen}
+            onOpen={onOpen}
+            onClose={onClose}
+          >
+            <PopoverTrigger>
+              <Button
+                isLoading={isTransferLoading}
+                isDisabled={isTransferLoading}
+                leftIcon={<IoIosSend />}
+              >
+                Send
+              </Button>
+            </PopoverTrigger>
```

Comment:
We should probably break some of these out into their own components, but not a big deal for now